src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -198,6 +198,10 @@ public function request(string $method, string $url, array $options = []): Respo
             $options['headers'][] = 'User-Agent: Symfony HttpClient (Native)';
         }
 
+        if (0 < $options['max_connect_duration']) {
+            $options['timeout'] = min($options['max_connect_duration'], $options['timeout']);
```

Comment:
This timeout should be used only for the fopen stage.
This means only the $context array below should get this logic to me.